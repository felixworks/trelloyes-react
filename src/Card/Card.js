import React from 'react';
import './Card.css';

function Card(props) {
    console.log('card props', props.listId);
    return (
        <div className="List-cards">
            <div className="Card">
                <h3>{props.title}</h3>
                <p>{props.content}</p>
                <button type="button" onClick={() => props.onClickDelete(props.cardId, props.listId)}>Delete</button> 
            </div>
        </div>
    )
}

export default Card;