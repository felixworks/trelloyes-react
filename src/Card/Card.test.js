import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Card from './Card.js';

describe('Card test', () => {
    it('renders without crashing', () => {
        const div = document.createElement("div");
    
        ReactDOM.render(<Card />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders card with correct info', () => {
        const tree = renderer
            .create(<Card />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
});