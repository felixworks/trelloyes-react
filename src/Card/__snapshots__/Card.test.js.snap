// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card test renders card with correct info 1`] = `
<div
  className="List-cards"
>
  <div
    className="Card"
  >
    <h3 />
    <p />
  </div>
</div>
`;
